crates/oxc_ast/src/trivia.rs

```diff
@@ -12,13 +12,20 @@ use oxc_span::Span;
 #[derive(Debug, Clone, Copy)]
 pub struct Comment {
     pub kind: CommentKind,
-    pub end: u32,
+    /// The span of the comment text (leading/trailing delimiters not included).
+    span: Span,
 }
 
 impl Comment {
     #[inline]
-    pub fn new(end: u32, kind: CommentKind) -> Self {
-        Self { kind, end }
+    pub fn new(start: u32, end: u32, kind: CommentKind) -> Self {
+        let span = Span::new(start, end);
+        Self { kind, span }
+    }
+
+    /// Return the span of the comment text (without delimiters).
+    pub fn span(&self) -> &Span {
```

Comment:
Shall we try adding `#[inline]` here? `.span()` is called a lot in the jsdoc builder, hence the performance regression (my guess).

Reply:
Definitely `#[inline]`. Also, `Span` is `Copy`, so we should also return `Span` not `&Span`.

Reply:
There is a `pub` to private change in here too (as @overlookmotel already pointed out below) which is possibly preventing some cross-crate inlining/optimization which was previously possible.
I'll try to expose it again in the next round and see if that fixes the benchmark regression.